### PR TITLE
Support for new PHP 8.3 reflection API

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/Adapter/ReflectionMethod.php [hunk 1]
```diff
@@ -34,6 +37,27 @@ public function __construct(private BetterReflectionMethod $betterReflectionMeth
         unset($this->class);
     }
 
+    public static function createFromMethodName(string $method): static
+    {
+        if (! str_contains($method, '::')) {
+            throw new CoreReflectionException('Argument #1 ($method) must be a valid method name');
+        }
+
+        [$className, $methodName] = explode('::', $method);
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionMethod.php]: What happens if a value like `foo::bar::baz` is given? I think we should have `2` as third argument here

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php [hunk 2]
```diff
@@ -265,6 +271,17 @@ public static function internalFunctionsProvider(): array
             array_filter(
                 $functionNames,
                 static function (string $functionName): bool {
+                    if (
+                        in_array($functionName, [
+                        // Missing in JetBrains/phpstorm-stubs
```
Comment (Ocramius) [inline test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php:276]: Are the stubs to be updated in `composer.lock`?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php (hunk 2 above)
Comment (kukulich) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php:276]: We use latest version currently. So we have to wait for new release.

File: src/Reflection/Adapter/ReflectionMethod.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/Adapter/ReflectionMethod.php]: Added 👍

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @kukulich!

**Pull request merged**